### PR TITLE
new TestSharedCollection, adding some more tests

### PR DESCRIPTION
**Pull request by DanSimon**
This is the beginnings of a simple testkit for colossus-metrics.  Don't think we'll need to break it into a separate project.

Also wrote some more tests.

### CODE REVIEW

File: colossus-metrics/src/main/scala/colossus/metrics/TestKit.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+package testkit
+
+import akka.testkit.TestProbe
+import scala.reflect.ClassTag
+
+
+/**
+ * A Fake shared collection for creating event collectors using a provided test probe
+ */
+class TestSharedCollection(probe: TestProbe) extends Collection[SharedLocality] {
```
Comment (nsauro) [inline colossus-metrics/src/main/scala/colossus/metrics/TestKit.scala]: Weren't we going to add the DeadMetricSystem here?

**Comment by nsauro** [comment]
:1

**Comment by nsauro** [comment]
after this is merged, I will merge in my branch with its single commit for modding the access modifier on SharedCollection

File: colossus-metrics/src/main/scala/colossus/metrics/TestKit.scala (hunk 1 above)
Comment (DanSimon) [inline colossus-metrics/src/main/scala/colossus/metrics/TestKit.scala]: Right now there actually is no class or object, just a method on `MetricSystem`, so I'll leave it be for now

**Pull request merged**